speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backend/SQLDatabaseStorage.cpp

```diff
@@ -604,6 +597,9 @@ void SQLDatabaseStorage::InsertDecisions(std::ifstream& p_inputFile, const int p
 /// @brief Close the connection to the database and clean up.
 void SQLDatabaseStorage::CloseDatabase()
 {
+    // Guard if database was already closed.
+    if (!m_statement && !m_connection) return;
```

Comment:
If only the statement has been closed but not the connection, this does not close the connection. These two should probably be checked separately?